Design/histogram_type.md

```diff
@@ -99,6 +236,12 @@ Furthermore, we will be able to decide whether or not we provide direct access t
 This direct access would not be identical to the current one, since it would rely on the new types for these data structures.
 
 
+## Python interface
+
+- There should be no major difficulties providing a Python interface.
```

Comment:
One thing I've just thought about is backwards-compatibility in the python interface. we currently expose numpy arrays via `readX()`. We're going to have to keep these around until we can deprecate their use. We should still introduce and expose the `Histogram` type an maybe give some talks and demos on why this should be the preferred method of interaction.